docs/scos/dev/upgradability-services/upgradability-guidelines/private-api-was-extended.md

```diff
@@ -0,0 +1,156 @@
+
+## Example of resolving the error by copying the form class to the project level
+
+```php
+<?php
+namespace Pyz\Zed\CustomerAccessGui\Communication\Form;
+
+use Spryker\Zed\Kernel\Communication\Form\AbstractType;
+
+class PyzCustomerAccessForm extends AbstractType
+{
+    ...
+}
+```
+
+## Code that causes an upgradability error: Extending a private API business model
+
+`to-provide` extends `to-proivde`. It is a private API.
+
+```php
+namespace Pyz\Zed\CustomerAccess\Business\CustomerAccess;
+
+use Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter as SprykerCustomerAccessFilter;
+
+class CustomerAccessFilter extends SprykerCustomerAccessFilter
+{
+    ...
+}
+```
+
+## Related error in the Evaluator output: Extending a private API business model
+
+```bash
+------------------------------------------------------------------------------------
+"PrivateApi:PrivateApiDependencyInBusinessModel: ""Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter" in "Pyz\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter""
+------------------------------------------------------------------------------------
+```
+
+## Example of resolving the error by copying the business model to the project level
+
+```php
+<?php
+namespace Pyz\Zed\CustomerAccess\Business\CustomerAccess;
+
+class PyzCustomerAccessFilter implements PyzCustomerAccessFilterInterface
+{
+    ...
+}
+```
+
+## Code that causes an upgradability error: Extending a private API dependency provider
+
+`to-provide` extends `to-proivde`. It is a private API.
+
+```php
+namespace Pyz\Yves\CheckoutPage;
+
+use Spryker\Yves\CheckoutPage\CheckoutPageDependencyProvider as SprykerCheckoutPageDependencyProvider;
+
+class CheckoutPageDependencyProvider extends SprykerCheckoutPageDependencyProvider
+{
+    /**
+     * ...
+     */
+    protected function getCustomerStepHandler(): ...
+    {
+        ...
+    }
+}
+```
+
+## Related error in the Evaluator output: Extending a private API dependency provider
+
+```bash
+------------------------------------------------------------------------------------
+"PrivateApi:MethodIsOverwritten: "Spryker\Yves\CheckoutPage\CheckoutPageDependencyProvider::getCustomerStepHandler" in "Pyz\Yves\CheckoutPage\CheckoutPageDependencyProvider""
+------------------------------------------------------------------------------------
+```
+
+#### Example of resolving the error by copying the dependency provider to the project level
+
+```php
+namespace Pyz\Yves\CheckoutPage;
+
+use Spryker\Yves\CheckoutPage\CheckoutPageDependencyProvider as SprykerCheckoutPageDependencyProvider;
+
+class CheckoutPageDependencyProvider extends SprykerCheckoutPageDependencyProvider
+{
+    /**
+     * ...
+     */
+    protected function getPyzCustomerStepHandler(): ...
+    {
+        ...
+    }
+}
+```
+{% info_block infoBox "" %}
+
```

Comment:
Please clarify what we are trying to explain in this note and what a reader should do with the code snippet.

Reply:
This has to go to a different documentation - https://docs.spryker.com/docs/scos/dev/architecture/module-api/definition-of-module-api.html 

"Plugin defining protected methods in dependency providers." 
It is already defined in that doc. 

If you want to - feel free to add more examples to that doc. 
But lets remove this example from the current documentation